doc/crypto/api/ops/key-encapsulation.rst

```diff
@@ -0,0 +1,350 @@
+
+    The shared output key that is produced by `PSA_ALG_ECIES_SEC1` is not suitable for use as an encryption key.
+    It must be used as an input to a key derivation operation to produce additional cryptographic keys.
+
+    .. subsection:: Compatible key types
+
+        | :code:`PSA_KEY_TYPE_ECC_KEY_PAIR(family)`
+        | :code:`PSA_KEY_TYPE_ECC_PUBLIC_KEY(family)` (encapsulaton only)
+
+        where ``family`` is a Weierstrass or Montgomery Elliptic curve family.
+        That is, one of the following values:
+
+        *   ``PSA_ECC_FAMILY_SECT_XX``
+        *   ``PSA_ECC_FAMILY_SECP_XX``
+        *   `PSA_ECC_FAMILY_FRP`
+        *   `PSA_ECC_FAMILY_BRAINPOOL_P_R1`
+        *   `PSA_ECC_FAMILY_MONTGOMERY`
+
+Key-encapsulation functions
+---------------------------
+
+.. function:: psa_encapsulate
+
+    .. summary::
+        Use a public key to generate a new shared secret key and associated encapsulation data.
+
+    .. param:: psa_key_id_t key
+        Identifier of the key to use for the encapsulation.
+        It must be a public key or an asymmetric key pair.
+        It must permit the usage `PSA_KEY_USAGE_ENCAPSULATE`.
+    .. param:: psa_algorithm_t alg
+        The key-encapsulation algorithm to use: a value of type `psa_algorithm_t` such that :code:`PSA_ALG_IS_KEY_ENCAPSULATION(alg)` is true.
+    .. param:: const psa_key_attributes_t * attributes
+        The attributes for the output key.
+        This function uses the attributes as follows:
+
+        *   The key type.
+            All key-encapsulation algorithms can output a key of type :code:`PSA_KEY_TYPE_DERIVE` or :code:`PSA_KEY_TYPE_HMAC`.
+            Key-encapsulation algorithms that produce a pseudo-random shared secret, can also output block-cipher key types, for example :code:`PSA_KEY_TYPE_AES`.
+            Refer to the documentation of individual key-encapsulation algorithms for more information.
+
+        The following attributes must be set for keys used in cryptographic operations:
+
+        *   The key permitted-algorithm policy, see :secref:`permitted-algorithms`.
+        *   The key usage flags, see :secref:`key-usage-flags`.
+
+        The following attributes must be set for keys that do not use the default volatile lifetime:
+
+        *   The key lifetime, see :secref:`key-lifetimes`.
+        *   The key identifier is required for a key with a persistent lifetime, see :secref:`key-identifiers`.
+
+        The following attributes are optional:
+
+        *   If the key size is nonzero, it must be equal to the size of the shared secret.
+
+        .. note::
+            This is an input parameter: it is not updated with the final key attributes.
+            The final attributes of the new key can be queried by calling `psa_get_key_attributes()` with the key's identifier.
+    .. param:: psa_key_id_t * output_key
+        On success, an identifier for the newly created shared output key.
```

Comment:
What about using "shared secret key" (this is the phrase used in FIPS203), as in the key with the shared secret?

I think 'shared output key' got into the text in order to disambiguate between the input key and the output key. In this API the input decapsulation key is `key`, so the output is `output_key`.In the key agreement API, the input key is called `private_key`, so the output is just `key`.

Perhaps use `decaps_key` as the input key, and then just `key` as the output - or maybe `shared_secret_key` to avoid confusion from truncated names?

Reply:
I think `output_key` is ok as the parameter name. It's the English description that I find unclear.